server/service/src/programs/program_indicator/query.rs

```diff
@@ -0,0 +1,121 @@
+use std::collections::HashMap;
+
+use repository::{
+    IndicatorColumnRow, IndicatorColumnRowRepository, IndicatorLineRow, IndicatorLineRowRepository,
+    Pagination, ProgramIndicatorFilter, ProgramIndicatorRepository, ProgramIndicatorRow,
+    ProgramIndicatorSort, RepositoryError, StorageConnection,
+};
+
+#[derive(Clone, Eq, PartialEq, Debug)]
+pub struct IndicatorLine {
+    pub line: IndicatorLineRow,
+    pub columns: Vec<IndicatorColumnRow>,
+}
+
+#[derive(Clone, Eq, PartialEq, Debug)]
+pub struct ProgramIndicator {
+    pub program_indicator: ProgramIndicatorRow,
+    pub lines: Vec<IndicatorLine>,
+}
+
+pub fn program_indicators(
+    connection: &StorageConnection,
+    pagination: Pagination,
+    sort: Option<ProgramIndicatorSort>,
+    filter: Option<ProgramIndicatorFilter>,
+) -> Result<HashMap<String, ProgramIndicator>, RepositoryError> {
+    let indicators = ProgramIndicatorRepository::new(connection).query(pagination, filter, sort)?;
+
+    let indicator_ids: Vec<String> = indicators
+        .clone()
+        .into_iter()
+        .map(|indicator| indicator.id)
+        .collect();
+
+    let all_indicator_line_rows =
+        IndicatorLineRowRepository::new(connection).find_many_by_indicator_ids(&indicator_ids)?;
+    let all_indicator_column_rows =
+        IndicatorColumnRowRepository::new(connection).find_many_by_indicator_ids(&indicator_ids)?;
+
+    let mut indicators_hash = HashMap::new();
+
+    for indicator in indicators {
+        let mut indicator_lines_hash: HashMap<String, IndicatorLine> = HashMap::new();
+        let indicator_id = indicator.id.clone();
+        let indicator_lines: Vec<IndicatorLineRow> = all_indicator_line_rows
+            .clone()
+            .into_iter()
+            .filter(|line| line.program_indicator_id == indicator_id)
```

Comment:
I would use partition: Can you partition btw, definitly not asking for this change, definitly not asking for this change, but here is how it could look: https://github.com/msupply-foundation/open-msupply/compare/5215-add-program-indicator-graphql-layer...5215-add-program-indicator-graphql-layer(Use-partition)

Not asking for this change !!

Reply:
Will get it in anyway (: Thanks Andrei!